sympy/solvers/solveset.py

```diff
@@ -1705,6 +1707,14 @@ def linear_eq_to_matrix(equations, *symbols):
     if hasattr(symbols[0], '__iter__'):
         symbols = symbols[0]
 
+    # setting default value for check_linearity
+    if 'check_linearity' not in options:
```

Comment:
`check_linearity = options.get('check_linearity', False)` is the way to set the flag without modifying the passed dictionary.

Reply:
I have implemented this and made the changes.